BrightFuturesTests/ResultTests.swift

```diff
@@ -151,26 +161,26 @@ class ResultTests: XCTestCase {
         XCTAssert(r.isFailure)
         XCTAssertEqual(r.error!.domain, "DivisionByZeroError")
     }
-    
-    func testRecoverNeeded() {
```

Comment:
I was looking into this as well and I think this is the main difference between BrightFutures' Result and antitypical/Result. We could try to file a PR for antitypical/Result with the `recover` and `recoverWith` functions.